components/StyledButton/index.js

```diff
@@ -2,13 +2,23 @@ import styled from "styled-components";
 
 export const StyledButton = styled.button`
   padding: 0.8rem;
+  margin: ${(props) =>
+    props.$variant === "delete" || props.$variant === "edit" ? "0" : "1rem 0"};
   border-radius: 0.6rem;
-  color: black;
-  text-decoration: none;
   font-weight: bold;
-  border: none;
+  color: white;
+  border: ${(props) =>
+    props.$variant === "delete" || props.$variant === "edit" ? " " : "none"};
   cursor: pointer;
-  background-color: ${(props) => {
-    return props.$variant === "delete" ? "white" : "lightsalmon";
-  }};
+  background: ${(props) =>
+    props.$variant === "delete" || props.$variant === "edit"
+      ? "var(--button-background)"
+      : "var(--button-gradient1)"};
+
+  &:hover {
+    background: ${(props) =>
+      props.$variant === "delete" || props.$variant === "edit"
+        ? "var(--button-hover-background)"
+        : "var(--button-gradient2)"};
+  }
```

Comment:
Since you make use of the `$variant` prop on many places here it is a good idea to use the styled-components' `css` function